data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostStateValidatorBalancesTest.java

```diff
@@ -0,0 +1,120 @@
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+
+class PostStateValidatorBalancesTest
+    extends AbstractMigratedBeaconHandlerWithChainDataProviderTest {
+
+  @BeforeEach
+  void setup() {
+    initialise(SpecMilestone.ALTAIR);
+    genesis();
+
+    setHandler(new PostStateValidatorBalances(chainDataProvider));
+  }
+
+  @Test
+  public void shouldGetSpecifiedValidatorBalancesFromState() throws Exception {
+    StubRestApiRequest request =
+        StubRestApiRequest.builder()
+            .metadata(handler.getMetadata())
+            .pathParameter("state_id", "head")
+            .build();
+    request.setRequestBody(List.of("1", "2"));
+
+    final Optional<ObjectAndMetaData<List<StateValidatorBalanceData>>> stateValidatorBalancesData =
+        chainDataProvider.getStateValidatorBalances("head", List.of("1", "2")).get();
+
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody()).isEqualTo(stateValidatorBalancesData.orElseThrow());
+  }
+
+  @Test
+  public void shouldGetAllValidatorBalancesFromState() throws Exception {
+    StubRestApiRequest request =
+        StubRestApiRequest.builder()
+            .metadata(handler.getMetadata())
+            .pathParameter("state_id", "head")
+            .build();
+
+    final Optional<ObjectAndMetaData<List<StateValidatorBalanceData>>> stateValidatorBalancesData =
+        chainDataProvider.getStateValidatorBalances("head", List.of()).get();
+
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody()).isEqualTo(stateValidatorBalancesData.orElseThrow());
+  }
+
+  @Test
+  void metadata_shouldHandle400() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_BAD_REQUEST);
+  }
+
+  @Test
+  void metadata_shouldHandle404() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_NOT_FOUND);
+  }
+
+  @Test
+  void metadata_shouldHandle500() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  void metadata_shouldHandle200() throws IOException {
+    List<StateValidatorBalanceData> stateValidatorBalanceData = new ArrayList<>();
+    for (int i = 0; i < 10; i++) {
+      stateValidatorBalanceData.add(
+          new StateValidatorBalanceData(UInt64.valueOf(i), dataStructureUtil.randomUInt64()));
+    }
+
+    ObjectAndMetaData<List<StateValidatorBalanceData>> responseData =
+        withMetaData(stateValidatorBalanceData);
+
+    final String data = getResponseStringFromMetadata(handler, SC_OK, responseData);
+    final String expected =
+        Resources.toString(
+            Resources.getResource(
+                GetStateValidatorBalancesTest.class, "getStateValidatorBalances.json"),
+            UTF_8);
```

Comment:
Could you please add a comment that the methods have the same expected response and that is why we are using the same json? Otherwise, in the future, we might think this was a mistake.